docs/rules/jsx-props-no-spreading.md

```diff
@@ -0,0 +1,24 @@
+# Disallow JSX props spreading (react/jsx-props-no-spreading)
+
+Enforces that there is no spreading for any JSX attribute. This enhances readability of code and hence leads to better maintainability.
+
+## Rule Details
+
+The following patterns are considered warnings:
+
+```jsx
+<App {...props} />
+<App {...props} some_other_prop={some_other_prop} />
+<div {...props}><App /></div>
+```
+
+The following patterns are **not** considered warnings:
+
+```jsx
+const {one_prop, two_prop} = props;
+<App one_prop={one_prop} two_prop={two_prop} />
+```
+
+## When Not To Use It
+
+If you are not using JSX or have lots of props to be passed or don't worry much about maintainability.
```

Comment:
"don't worry much about maintainability" is a bit snarky and passive aggressive. Let's reword.

Reply:
Yes, very much agree. Updated